code/game/machinery/washing_machine.dm

```diff
@@ -26,286 +20,227 @@
+		if(!U)
+			for(var/T in typesof(/obj/item/clothing/under/rank))
+				var/obj/item/clothing/under/rank/R = T
+				if(wash_color == initial(R.item_color))
+					U = R
+					break
+		if(U)
+			item_state = initial(U.item_state)
+			icon_state = initial(U.icon_state)
+			item_color = wash_color
+			name = initial(U.name)
+			desc = "The colors are a bit dodgy."
+			can_adjust = initial(U.can_adjust)
+			if(!can_adjust && adjusted) //we deadjust the uniform if it's now unadjustable
+				toggle_jumpsuit_adjust()
+
+/obj/item/clothing/gloves/color/machine_wash(obj/machinery/washing_machine/WM)
+	if(WM.color_source)
+		var/wash_color = WM.color_source.item_color
+		for(var/T in typesof(/obj/item/clothing/gloves/color))
+			var/obj/item/clothing/gloves/color/G = T
+			if(wash_color == initial(G.item_color))
+				item_state = initial(G.item_state)
+				icon_state = initial(G.icon_state)
+				item_color = wash_color
+				name = initial(G.name)
+				desc = "The colors are a bit dodgy."
+				break
+
+/obj/item/clothing/shoes/sneakers/machine_wash(obj/machinery/washing_machine/WM)
+	if(chained)
+		chained = 0
+		slowdown = SHOES_SLOWDOWN
+		new /obj/item/weapon/restraints/handcuffs(loc)
+	if(WM.color_source)
+		var/wash_color = WM.color_source.item_color
+		for(var/T in typesof(/obj/item/clothing/shoes/sneakers))
+			var/obj/item/clothing/shoes/sneakers/S = T
+			if(wash_color == initial(S.item_color))
+				icon_state = initial(S.icon_state)
+				item_color = wash_color
+				name = initial(S.name)
+				desc = "The colors are a bit dodgy."
+				break
+
+/obj/item/weapon/bedsheet/machine_wash(obj/machinery/washing_machine/WM)
+	if(WM.color_source)
+		var/wash_color = WM.color_source.item_color
+		for(var/T in typesof(/obj/item/weapon/bedsheet))
+			var/obj/item/weapon/bedsheet/B = T
+			if(wash_color == initial(B.item_color))
+				icon_state = initial(B.icon_state)
+				item_color = wash_color
+				name = initial(B.name)
+				desc = "The colors are a bit dodgy."
+				break
+
+/obj/item/clothing/head/soft/machine_wash(obj/machinery/washing_machine/WM)
+	if(WM.color_source)
+		var/wash_color = WM.color_source.item_color
+		for(var/T in typesof(/obj/item/clothing/head/soft))
+			var/obj/item/clothing/head/soft/H = T
+			if(wash_color == initial(H.item_color))
+				icon_state = initial(H.icon_state)
+				item_color = wash_color
+				name = initial(H.name)
+				desc = "The colors are a bit dodgy."
+				break
+
+
+/obj/machinery/washing_machine/relaymove(mob/user)
+	container_resist(user)
+
+/obj/machinery/washing_machine/container_resist(mob/user)
+	if(!busy)
+		add_fingerprint(user)
+		open_machine()
+
 
 
 /obj/machinery/washing_machine/update_icon()
-	icon_state = "wm_[state]0"
+	overlays.Cut()
+	if(busy)
+		icon_state = "wm_running_[bloody_mess]"
+	else if(bloody_mess)
+		icon_state = "wm_[state_open]_blood"
+	else
+		var/full = contents.len ? 1 : 0
+		icon_state = "wm_[state_open]_[full]"
+	if(panel_open)
+		overlays += image(icon, icon_state = "wm_panel")

```

Comment:
by using an overlay for the open panel, we divide by two the number of sprites needed.